src/rust/pt-br.goldenmangas/src/parser.rs

```diff
@@ -0,0 +1,292 @@
+use aidoku::{
+	error::Result,
+	helpers::substring::Substring,
+	prelude::*,
+	std::{
+		net::{HttpMethod, Request},
+		String, StringRef, Vec,
+	},
+	Chapter, DeepLink, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaStatus,
+	MangaViewer, Page,
+};
+
+use crate::helper::{append_domain, get_chapter_number, get_search_url, manga_status};
+
+pub fn parse_manga_list(
+	base_url: String,
+	filters: Vec<Filter>,
+	page: i32,
+) -> Result<MangaPageResult> {
+	let mut included_tags: Vec<String> = Vec::new();
+	let mut title: String = String::new();
+	let status_options = ["", "ativo", "completo"];
+	let sort_options = [
+		"", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q",
+		"R", "S", "T", "U", "V", "W", "X", "Y", "Z",
+	];
+	let mut sort_letter: String = String::new();
+	let mut status: String = String::new();
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				title = filter.value.as_string()?.read();
+			}
+			FilterType::Genre => match filter.value.as_int().unwrap_or(-1) {
+				1 => included_tags.push(filter.object.get("id").as_string()?.read()),
+				_ => continue,
+			},
+
+			FilterType::Select => {
+				let index = filter.value.as_int().unwrap_or(-1) as usize;
+				match filter.name.as_str() {
+					"Status" => status = String::from(status_options[index]),
+					"ByLetter" => sort_letter = String::from(sort_options[index]),
+					_ => continue,
+				}
+			}
+			_ => continue,
+		};
+	}
+	let url = get_search_url(base_url, title, page, included_tags, status, sort_letter);
+	parse_manga_listing(url, String::new(), page)
+}
+
+pub fn parse_manga_listing(
+	base_url: String,
+	listing_name: String,
+	page: i32,
+) -> Result<MangaPageResult> {
+	let list_url = if !listing_name.is_empty() {
+		match listing_name.as_str() {
+			"Últimas Atualizações" => format!("{base_url}/index.php?pagina={page}"),
+			_ => format!("{base_url}/mangabr&pagina={page}"),
+		}
+	} else {
+		base_url.clone()
+	};
+	let mut _count = 0;
+	let mut mangas: Vec<Manga> = Vec::new();
+	let html = Request::new(&list_url, HttpMethod::Get).html()?;
+	if listing_name.is_empty() {
+		for manga in html.select(".mangas a").array() {
+			let manga_node = manga.as_node().expect("Failed to get manga as node");
+			let title = manga_node.select("h3").text().read();
+			let url_path = manga_node.attr("href").read();
+			let url = append_domain(base_url.clone(), url_path.clone());
+			let id = String::from(url_path.substring_after("mangabr/").unwrap_or_default());
+			let cover = append_domain(
+				String::from("https://goldenmangas.top"),
```

Comment:
```suggestion
				base_url.clone(),
```

Reply:
It will break search results from [parse_manga_list](https://github.com/Skittyblock/aidoku-community-sources/blob/2040b45e7677a88f8bbc6d4bea3e29addc6a4db9/src/rust/pt-br.goldenmangas/src/parser.rs#L51) as I'm calling the same function to list manga.